### ISSUE TITLE
Missing Screenshot attachment in VSTS but see a path in .trx

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit3-vs-adapter#526]
@charleshamel73 commented on [Mon Jul 02 2018](https://github.com/nunit/nunit/issues/2918)

Hello,
I am currently working with:
VSTestV2
VSTS 
Visual Studio 2017
NUnit 3.10.*

While using the `TestContext.AddTestAttachment(fullFilePath)` function, I see an attachment within the .trx file but that attachment isn't shown within test attachments page. I have tried to fix the path to be the test run settings but nothing is working. I have attached a screenshot of the Code and the Attachments Page without the attached screenshot
```csharp
 if (TestContext.CurrentContext.Result.Outcome != ResultState.Success)
 {
     String baseTestDir = TestContext.CurrentContext.WorkDirectory;
     if ( environment.HasEnvironmentValue("SYSTEM_DEFAULTWORKINGDIRECTORY"))
     {
         baseTestDir = String.Format("{0}\\TestResults",environment.GetEnvironmentValue("SYSTEM_DEFAULTWORKINGDIRECTORY"));
     }
     String fullFilePath = String.Format("{0}\\{1}.png", baseTestDir, TestContext.CurrentContext.Test.FullName);
     log.Info(String.Format("Test has failed. Taking Screenshot and writing it to '{0}'", fullFilePath));
     Screenshot screenshot = driver.GetScreenshot();
     screenshot.SaveAsFile(fullFilePath, ScreenshotImageFormat.Png);
     TestContext.AddTestAttachment(fullFilePath);
 }
```

Anyone have any idea what I might be missing?

---

@charleshamel73 commented on [Mon Jul 02 2018](https://github.com/nunit/nunit/issues/2918#issuecomment-401941565)

Update* I did found out that VSTestV2 has a very specific folder to which it looks for attachements in 
```
2018-07-02T20:16:31.6086833Z ##[debug]Reading test results from file 'D:\a\1\s\TestResults\VssAdministrator_*_2018-07-02_20_15_38.trx'.
2018-07-02T20:16:31.6117584Z ##[debug]Setting run start and finish times.
2018-07-02T20:16:31.6166434Z ##[debug]Attachment location: D:\a\1\s\TestResults\VssAdministrator_*_2018-07-02_20_15_38\In.
2018-07-02T20:16:31.6314450Z ##[debug]Total test results: 4.
```

---

@mikkelbu commented on [Mon Jul 02 2018](https://github.com/nunit/nunit/issues/2918#issuecomment-401951772)

Hi @charleshamel73. I don't think I can help you, as I know very little about VSTestV2 and VSTS, but I just wanted you to tell you that I've edited the code blocks to make them read nicer (by using [fenced code blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/)).

---

@CharliePoole commented on [Mon Jul 02 2018](https://github.com/nunit/nunit/issues/2918#issuecomment-401971593)

@charleshamel73 This seems to be an issue for the NUnit 3 VS adapter, rather than the framework. We can move it there so the right people see it if you agree.

---

@charleshamel73 commented on [Tue Jul 03 2018](https://github.com/nunit/nunit/issues/2918#issuecomment-402189581)

Yes That would be great if you could. Even if this is just a problem on my end, i would like other users to at least be able to refer to this thread and be able to find a solution

**Comment by OsirisTerje** [comment]
Can this be a duplicate of #494?   @charleshamel73  Would be nice if you could check out the latest development package from our myget feed which has 494 fixed.  See https://www.myget.org/feed/nunit/package/nuget/NUnit3TestAdapter  , version 3.11.0-dev-00894

**Comment by Kira-Lappo** [comment]
@CharliePoole did you tried to use jpeg instead of png? As I understand, trx file contains correct path to the file, right? May be the reason is file size limitation.

**Comment by CharliePoole** [comment]
@Kira-Lappo Wrong guy! I think you meant your comment for @charleshamel73

**Comment by Kira-Lappo** [comment]
@CharliePoole right), I'm sorry for disturbing
@charleshamel73, that question was for you : 
 did you tried to use jpeg instead of png? As I understand, trx file contains correct path to the file, right? May be the reason is file size limitation.

**Comment by Kira-Lappo** [comment]
I created build in VSTS and run test which attaches png. The png appeared in test result attachments.
> 'D:\a\1\s\TestResults\VssAdministrator_*_2018-07-02_20_15_38.trx'.

Instead of * there should be a name of test machine, it's intresting, why dotnet test puts the asterisk here. For example, I got 
> 2018-07-09T18:10:46.9444592Z Results File: D:\a\1\s\TestResults\VssAdministrator_factoryvm-az493_2018-07-09_18_10_45.trx

**Comment by srinukamma79** [comment]
I created build in VSTS and run test which attachment html. The html attachments are showing but preview is not coming. Please help me.
![vsts_nunit](https://user-images.githubusercontent.com/44565348/47634676-5dbc9a80-db78-11e8-9e49-9460e479a68e.JPG)

**Comment by OsirisTerje** [comment]
@srinukamma79 @charleshamel73  @Kira-Lappo     The #494 bug was fixed in 3.10, so that is not the issue.  Then check the response to this https://stackoverflow.com/questions/39905758/where-is-system-defaultworkingdirectory-set-for-builds-in-tfs2015  about the the setting of the SYSTEM_DEFAULTWORKINGDIRECTORY.   If nothing of this helps, please raise this question to MS at https://github.com/microsoft/vsts-tasks  (or https://developercommunity.visualstudio.com/spaces/21/index.html  or https://developercommunity.visualstudio.com/spaces/22/index.html )  .  It doesn't look like this in the adapter.

**Comment by srinukamma79** [comment]
> @srinukamma79 @charleshamel73 @Kira-Lappo The #494 bug was fixed in 3.10, so that is not the issue. Then check the response to this https://stackoverflow.com/questions/39905758/where-is-system-defaultworkingdirectory-set-for-builds-in-tfs2015 about the the setting of the SYSTEM_DEFAULTWORKINGDIRECTORY. If nothing of this helps, please raise this question to MS at https://github.com/microsoft/vsts-tasks (or https://developercommunity.visualstudio.com/spaces/21/index.html or https://developercommunity.visualstudio.com/spaces/22/index.html ) . It doesn't look like this in the adapter.

Below is my process. 
For my selenium automation we will generate nunit xml with html attachments. The vsts is showing html attachment files but not showing preview. please suggest if any format we need to follow for attachments to get preview.

**Comment by OsirisTerje** [comment]
@srinukamma79  PLease report this issue at the links given above.  This is not an adapter issue, it is a vstest issue.
